### PR TITLE
Refactor Changes

### PR DESCRIPTION
**Pull request by msahilan07**
## What

Refactoring Changes

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/test/java/gov/uk/courtdata/hearing/processor/HearingWQProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @Before
+    public void setUp() throws Exception {
+
+        MockitoAnnotations.initMocks(this);
+
+
+
+    }
+
+
```
Comment (msahilan07) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/hearing/processor/HearingWQProcessorTest.java:28]: Update the test case with assert statements.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/laastatus/builder/RepOrderUpdateMessageBuilder.java [hunk 2]
```diff
@@ -81,39 +81,13 @@ private DefenceOrganisation mapDefenceOrganisation(CaseDetails caseDetails) {
     private Organisation findSolicitorDetails(SolicitorMAATDataEntity solicitorMAATDataEntity) {
 
         return Organisation.builder()
-                //TODO: to be enabled when CDA handled invalid data.
                 //.address(mapAddress(solicitorMAATDataEntity))
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/laastatus/builder/RepOrderUpdateMessageBuilder.java:84]: if this is not required then could remove these two lines.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java [hunk 3]
```diff
@@ -16,6 +17,7 @@
 
 @RequiredArgsConstructor
 @Component
+@Slf4j
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java:20]: Maybe we don't need this annotation here.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/RepOrderCPInfoProcessor.java [hunk 4]
```diff
@@ -21,7 +21,8 @@ public void process(CourtDataDTO courtDataDTO) {
         final CaseDetails caseDetails = courtDataDTO.getCaseDetails();
         Optional<RepOrderCPDataEntity> repOrderEntity = repOrderDataRepository.findByrepOrderId(caseDetails.getMaatId());
 
-        RepOrderCPDataEntity repOrder = repOrderEntity.get();
+        RepOrderCPDataEntity repOrder = repOrderEntity.orElse(null);
+        assert repOrder != null;
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/RepOrderCPInfoProcessor.java:25]: I personally think we should avoid 'assert'.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/util/DateUtil.java [hunk 5]
```diff
@@ -4,6 +4,10 @@
 
 public final class DateUtil {
 
+    private DateUtil() {
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/util/DateUtil.java:7]: Could use constructor annotation here.

File: maat-court-data-api/src/test/java/gov/uk/courtdata/hearing/processor/HearingWQProcessorTest.java (hunk 1 above)
Comment (msahilan07) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/hearing/processor/HearingWQProcessorTest.java:28]: Done!! ###

File: maat-court-data-api/src/main/java/gov/uk/courtdata/laastatus/builder/RepOrderUpdateMessageBuilder.java (hunk 2 above)
Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/laastatus/builder/RepOrderUpdateMessageBuilder.java]: Done

File: maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java (hunk 3 above)
Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java]: As discussed, This is private Cons, Dont want to over engineer it with combinations of  annotations.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/RepOrderCPInfoProcessor.java (hunk 4 above)
Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/RepOrderCPInfoProcessor.java]: Done!!

File: maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java (hunk 3 above)
Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/link/processor/OffenceInfoProcessor.java]: Done

File: maat-court-data-api/src/main/java/gov/uk/courtdata/util/DateUtil.java (hunk 5 above)
Comment (msahilan07) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/util/DateUtil.java:7]: As discussed, This is private Cons, Dont want to over engineer it with combinations of annotations

**Pull request merged**